### PR TITLE
Feature/add license key store

### PR DESCRIPTION
**Pull request by raymondfeng**
/to @ritch 
/cc @rmg @anthonyettinger 

The PR adds local key store for licenses and enables license push upon pm actions.

It depends on https://github.com/strongloop/strong-mesh-client/pull/19

### CODE REVIEW

**Comment by raymondfeng** [comment]
@ritch ping

File: arc-api/server/models/key-store.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @constructor
+ */
+function KeyStore(options) {
+  options = options || {};
+  this.root = options.root || path.join(homeDir, '.strongloop');
+  this.perUser = options.perUser || false;
+}
+
+KeyStore.prototype.getStoreFile = function(userId) {
+  var name = 'licenses.json';
```
Comment (ritch) [inline arc-api/server/models/key-store.js]: Would be nice to make this configurable...

File: arc-api/server/models/subscription.js [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+function getAccessToken(req) {
+  var token = req.query.access_token;
+  if (token) {
+    return token;
+  }
+  token = req.get('authorization');
+  if (typeof token === 'string') {
+    // Add support for oAuth 2.0 bearer token
+    // http://tools.ietf.org/html/rfc6750
+    if (token.indexOf('Bearer ') === 0) {
```
Comment (ritch) [inline arc-api/server/models/subscription.js]: Why are you duplicating this from loopback?

File: arc-api/server/models/subscription.js [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        store.save({licenses: body, modified: new Date()}, userId, function(err) {
+          cb(err, body);
+        });
+      }
+    });
+  };
+
+  function reloadSubscriptions(userId, accessToken, result, cb) {
+    Subscription.getSubscriptions(userId, 'online', {
+      query: {access_token: accessToken}
```
Comment (ritch) [inline arc-api/server/models/subscription.js]: Maybe force `accessToken` to be atleast `null`... I think we remove undefined properties and may match more than you want... Although I'm not 100% sure.

File: client/www/scripts/modules/common/arc-services.js [hunk 4]
```diff
@@ -178,6 +178,207 @@ module.factory(
     return R;
   }]);
 
+/**
```
Comment (ritch) [inline client/www/scripts/modules/common/arc-services.js]: Why is this code committed :(

**Comment by ritch** [comment]
Only a few minor comments. Feel free to address or not. LGTM :+1:

File: arc-api/server/models/key-store.js (hunk 1 above)
Comment (rmg) [inline arc-api/server/models/key-store.js]: While I agree on principle, the decision to make other modules depend on `~/.strongloop/licenses.json` being the path is somewhat mutually exclusive with making it configurable.

See https://github.com/strongloop/strong-arc/issues/1069#issuecomment-77671719

File: arc-api/server/models/subscription.js (hunk 2 above)
Comment (raymondfeng) [inline arc-api/server/models/subscription.js]: I was trying to use the same code from AccessToken but this function is not exported.

File: arc-api/server/models/subscription.js (hunk 3 above)
Comment (raymondfeng) [inline arc-api/server/models/subscription.js]: This is not for DB, it's faking the `req` object.

File: client/www/scripts/modules/common/arc-services.js (hunk 4 above)
Comment (raymondfeng) [inline client/www/scripts/modules/common/arc-services.js]: Not sure. It was there before.

**Pull request merged**